python/scripts/cirkopt.py

```diff
@@ -0,0 +1,167 @@
+            description="Generate plots showing search space",
+        )
+        # prefixing the argument with -- means it's optional
+        parser.add_argument(
+            "--param",
+            help="Paramater to sweep, e.g.: width",
+            default="width",
+        )
+        parser.add_argument(
+            "--range",
+            help=(
+                "Range of numbers to sweep over, MATLAB style (start:increment:end)"
+                " e.g.: 200e-9:100e-9:1000e-9"
+            ),
+            default="200e-9:100e-9:1000e-9",
+        )
+        parser.add_argument(
+            "--outpin",
+            help="Name of pin to get data from, e.g.: Y",
+            default="Y",
+        )
+        parser.add_argument(
+            "--outindex",
+            help="Index of value from LDB table to show, comma separated, e.g.: 0,1",
+            default="0,1",
+        )
+        parser.add_argument(
+            "--outdir",
+            help=(
+                "Directory to place results in, e.g. graphs. "
+                "Does not include generated netlists or LDB library."
+            ),
+            default=os.path.join(PYTHON_DIRECTORY, "out"),
+        )
+        parser.add_argument(
+            "--workdir",
+            help=(
+                "Directory to place generated netlists. "
+                "Must match settings in Liberate configeration files. "
+            ),
+            default=os.path.join(LIBERATE_DIRECTORY, "netlist_wrk"),
+        )
+        parser.add_argument(
+            "--netlist",
+            help=(
+                "Path to reference netlist to modify. "
+                "Must match settings in Liberate configeration file."
+            ),
+            default=os.path.join(LIBERATE_DIRECTORY, "netlist_ref/INVX1.sp"),
+        )
+        parser.add_argument(
+            "--ldb",
+            help=(
+                "Path to LDB library generated by Liberate. "
+                "Must match settings in Liberate configeration files."
+            ),
+            default=os.path.join(LIBERATE_DIRECTORY, "lib/example_tt_1.0_70_nldm.lib"),
+        )
+        parser.add_argument(
+            "--debug",
+            help="Print lots of debugging statements",
+            action="store_const",
+            dest="loglevel",
+            const=DEBUG,
+            default=WARNING,
+        )
+        parser.add_argument(
+            "--verbose",
+            help="Be verbose",
+            action="store_const",
+            dest="loglevel",
+            const=INFO,
+        )
+
+        # now that we're inside a subcommand, ignore the first
+        # TWO argvs, ie the command (cirkopt) and the subcommand (explore)
+        args = parser.parse_args(sys.argv[2:])
+
+        logging.basicConfig(
+            format="%(levelname)s (%(asctime)s): %(message)s",
+            datefmt="%m/%d/%Y %I:%M:%S %p",
+            level=args.loglevel,
+        )
+
+        param_dict = {
```

Comment:
nit: could do `{str(param).lower(): param for param in Param}` doesn't matter here since enum likely will never change. But if the enum was to get new values its a pain to go find a replace all hardcoded usages

Reply:
I didn't know the params have str values, is this a universal python thing because it certainly wouldn't work in c++ for example

Reply:
Nope, but i overrode the `__str__` method on the `Param` enum. So it works for us